app/Controllers/ProfileController.php

```
@@ -159,8 +162,18 @@ public function update(Request $request, Response $response, mysqli $db): Respon
             $locale = ($locale !== '' && isset($availableLocales[$locale])) ? $locale : null;
         }
 
-        // Validate required fields
-        if (empty($nome) || empty($cognome)) {
+        // Validate required fields. The surname follows the admin toggle
+        // (issue #255); when optional it is stored as '' — the column is NOT
+        // NULL by design so display paths that CONCAT nome+cognome stay whole.
+        if (empty($nome) || ($cognome === '' && \App\Support\RegistrationFields::isRequired('cognome'))) {
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Support\RegistrationFields' in method 'update'. (undefined)

(StaticAccess)